Content.Server/StationEvents/Events/RandomSentience.cs

```diff
@@ -49,13 +51,27 @@ public override void Startup()
         var kind1 = groupList.Count > 0 ? groupList[0] : "???";
         var kind2 = groupList.Count > 1 ? groupList[1] : "???";
         var kind3 = groupList.Count > 2 ? groupList[2] : "???";
-        _chatManager.DispatchStationAnnouncement(
-            Loc.GetString("station-event-random-sentience-announcement",
-                ("kind1", kind1), ("kind2", kind2), ("kind3", kind3), ("amount", groupList.Count),
-                ("data", Loc.GetString($"random-sentience-event-data-{_random.Next(1, 6)}")),
-                ("strength", Loc.GetString($"random-sentience-event-strength-{_random.Next(1, 8)}"))),
-                playDefaultSound: false,
-                colorOverride: Color.Gold
-        );
+
+        var _stationSystem = EntitySystem.Get<StationSystem>();
+        foreach (var target in targetList)
+        {
+            var victimOwningStation = _stationSystem.GetOwningStation(target.Owner);
+            if (victimOwningStation != null)
```

Comment:
You might be better off just making these Try statements with the station as an out param seeing how many there are. Also hashset again.